### ISSUE TITLE
`keyword-spacing` should disallow tabs (or have an option to do so)

### ISSUE THREAD

**Issue by pitaj** [eslint/eslint#8662]
**ESLint version:** 3.19.0
**Problem I want to solve:** accidental tabs after `var` and `let`

**My proposal:**
Currently the `keyword-spacing` rule has two possible values for each sub-option: `true` and `false`. I propose adding a `"space"` possibility here to specify that these keywords must have a space before or after them. 

Current usage, requiring an whitespace before and after every keyword
```json
{
  "keyword-spacing": ["error", {
    "before": true,
    "after": true
  }]
}
```

Proposed usage, requiring any whitespace before but only spaces after it
```json
{
  "keyword-spacing": ["error", {
    "before": true,
    "after": "space"
  }]
}
```

**Motivation:**
In a style with 4-character-wide tabs, the following two lines look identical
```js
var thing = "stuff";
var/*tab*/thing = "stuff";
```

The proposed tertiary option for the `keyword-spacing` rule would catch the second line above, preventing this common, though inconsequential, typo.

**Comment by ilyavolodin** [comment]
The same think can be acheaved by enabling `no-mixed-spaces-and-tabs` rule. Each rule has just one responsibility in ESLint. `keyword-spacing` job is to check that there's a space around keywords (doesn't matter what type), and `no-mixed-spaces-and-tabs` is there to check that you don't have accidental tabs through-out your code.

**Comment by pitaj** [comment]
`no-mixed-spaces-and-tabs` does not do that.

**Comment by ilyavolodin** [comment]
Hmm... That feels like an oversight for `no-mixed-spaces-and-tabs` maybe we should add an option there to disallow tabs anywhere?

**Comment by pitaj** [comment]
Well the rule `no-tabs` already disallows tabs everywhere, but I want to use tabs for indentation while simultaneously disallowing spaces after keywords.

**Comment by platinumazure** [comment]
I think enhancing `no-tabs` to support tolerating tabs at the beginning of the line but forbidding them elsewhere makes a lot of sense.

**Comment by ilyavolodin** [comment]
Ah, sorry, I was talking about `no-mixed-spaces-and-tabs` but I was actually thinking about `no-tabs`. Yes, I agree with @platinumazure adding an option to `no-tabs` seems like the best way to solve this problem.

**Comment by kaicataldo** [comment]
@platinumazure Agreed!

@pitaj Would you be willing to make a PR if this is accepted?

**Comment by pitaj** [comment]
Unfortunately I know nothing about how ESLint works internally and wouldn't know where to start. If you have some good resources I might go from there if I'm bored at some point this week.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
